data-processing/entities/sentences/extractor.py

```diff
@@ -67,24 +79,26 @@ def parse(self, tex_path: str, tex: str) -> Iterator[Sentence]:
             #    ex: "hello. world. 1) item one. 2) item two. 3) item three" or "hello!!! world."
             #    sol: set indexes manually using string search + sentence length
             # 3. pysbd uses reserved characters for splitting sentences
-            #    ex: see PSYDB_RESERVED_CHARACTERS list.
+            #    ex: see PYSBD_RESERVED_CHARACTERS list.
             #    sol: throw a warning if the sentence contains any of these characters.
-            if len(sentence.sent) > 1000:
-                logging.warning('Exceptionally long sentence (length %d), this might indicate the sentence extractor failed to properly split text into sentences.', len(sentence.sent))
+            if len(sentence) > 1000:
+                logging.warning('Exceptionally long sentence (length %d), this might indicate the sentence extractor failed to properly split text into sentences.', len(sentence))
 
-            real_start = plaintext.find(sentence.sent, length_so_far_in_plain_text)
-            real_end = real_start + len(sentence.sent)
-            if real_start not in plaintext_to_tex_offset_map or real_end not in plaintext_to_tex_offset_map:
-                logging.warning('A sentence boundary was incorrect for sentence %s. This is probably an issue with pysbd. Skipping sentence in extractor.', sentences.sent)
+            plaintext_start = plaintext.find(sentence, length_so_far_in_plain_text)
+            plaintext_end = plaintext_start + len(sentence)
+            if plaintext_start not in plaintext_to_tex_offset_map or plaintext_end not in plaintext_to_tex_offset_map:
+                logging.warning('A sentence boundary was incorrect for sentence %s. This is probably an issue with pysbd. Skipping sentence in extractor.', sentence)
                 continue
+            if plaintext_start - 500 > length_so_far_in_plain_text:
+                logging.warning('Sentence boundary start for sentence %s was %d characters a head of the previous sentence, this might indicate the sentence extractor failed to properly split text.', sentence, plaintext_start - length_so_far_in_plain_text)
```

Comment:
```suggestion
                logging.warning('Sentence boundary start for sentence %s was %d characters ahead of the previous sentence, this might indicate the sentence extractor failed to properly split text.', sentence, plaintext_start - length_so_far_in_plain_text)
```